### ISSUE TITLE
Placeholder is not keeping image ratio

### ISSUE THREAD

**Issue by gaborpeter** [tuupola/jquery_lazyload#341]
Hello!

My issue that I have images which have a width and height attribute given, but the placeholder is always 1:1 ratio as the placeholder image is 1x1 px.

The placeholder is overridable via the config parameter, but the images loaded on the site are not the same ratio, so one fix sized/ratio image is not fitting to the case.

The reason why it is causing an issue, as the site is also using masonry (in my case isotope.js), thus to avoid flicking when the proper ratio image is loaded, I want to preload placeholder with the proper ratio. Backend solution is not acceptable, as I want to avoid any additional browser requests or data transfer.

My idea, is to keep the placeholder config parameter by default to null, and modify the part of the code which is adding the placeholder when SRC is not given based on the available width and height information. 

```javascript
      /* If no src attribute given use data:uri. */
      if ($self.attr("src") === undefined || $self.attr("src") === false) {
        if ($self.is("img")) {
          if (settings.placeholder != null) {
            $self.attr("src", settings.placeholder);
          } else {
            var width = 1;
            var height = 1;
            if ($self.attr("width")) {
              width = $self.attr("width");
            }
            if ($self.attr("height")) {
              height = $self.attr("height");
            }
            $self.attr("src", "data:image/svg+xml;charset=utf-8,%3Csvg xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg' viewBox%3D'0 0 "+width+" "+height+"'%2F%3E");
          }
        }
      }
```

We also could thing about one more parameter which is 'placeholderColor'.

Worked for me, what do you think about this?

**Comment by jogaco** [comment]
Did you try to style in line?
`<img class="lazy" data-original="img/example.jpg" width="640" height="480" style="width: 640px; height: 480px">`

**Comment by gaborpeter** [comment]
Yes, it doesn't work either, here is a fiddle demonstrating the 3 scenarios 
https://jsfiddle.net/8q939jgf/

**Comment by gaborpeter** [comment]
Here's a live demo with the SVG solution. http://www.canoeicf.com/social-wall
If you load the page and disable JS, you can see image placeholders will eat up exactly the amount of space needed on all breakpoints, as they are 100% wide fitting the container, their height is set to auto. This really help to properly calculate the masonry already at page load and avoid isotope recalculation and flickering effects.

**Comment by jogaco** [comment]
I do not need a placeholder. The empty space is fine for me. The problem I was having is a repaint of a table in Firefox while lazy was loading the images: rows shrinking and going back to their size. I can avoid this by adding the inline style. This could be mentioned in the README by the way.

**Comment by gaborpeter** [comment]
I understand your case, however according to me as a general approach inline styles are not necessarily nice when the tool could out-of-the-box support the use case.

The library already has a placeholder function, which is adding the src tag with data: this would be only an improvement of this, to support image ratios and avoid some resizing / flickering.

**Comment by jogaco** [comment]
Check out https://github.com/verlok/lazyload. Seems like a more active version of this plugin, in plain javascript.

**Comment by tuupola** [comment]
You can either use placeholder images with same dimensions as the real image. See for example [this example](https://appelsiini.net/2017/trilateration-with-n-points/). Alternatively you can use html background images. This way it does not matter what dimensions the placeholder is.
